packages/video_player/video_player/lib/video_player.dart

```diff
@@ -378,6 +379,10 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
 
   Future<ClosedCaptionFile>? _closedCaptionFileFuture;
   ClosedCaptionFile? _closedCaptionFile;
+  List<Caption>? _sortedCaptions;
+
+  /// The sorted list of captions from the closed caption file.
+  List<Caption>? get sortedCaptions => _sortedCaptions;
```

Comment:
Why is new public API needed? I don't see any implementation code using this.

Reply:
To be able to access that list in testing, as I mentioned in the last comment I couldn't make it private and available for testing, 

If the test is not needed I can remove it with the test

Reply:
i removed it and the test.

Reply:
If something needs to be visible only for testing, it should be annotated `@visibleForTesting`.

It's not clear to me why even that would be necessary though; API to get the current caption is already public, and I would expect that it should be straightforward to construct a case where getting the caption for some time would be incorrect unless the captions are sorted internally.

Reply:
```
abstract class ClosedCaptionFile {
  /// The full list of captions from a given file.
  ///
  /// The [captions] will be in the order that they appear in the given file.
  List<Caption> get captions;
}

```

its only a getter so calling sort on it doesnt sort the original list, this is why i copied the list and sorted it and the public method was for the testing part only, if the test is not needed the public method is not needed

but `_sortedCaptions` is needed

Reply:
> this is why i copied the list and sorted it
> [...]
> but `_sortedCaptions` is needed

I'm not sure what part of my comments led you to believe I don't understand `_sortedCaptions`. My comment is about the new public API.

> the public method was for the testing part only, if the test is not needed the public method is not needed

I still don't understand why you are presenting a binary choice between adding new public API and having testing. Whether or not the correct caption is fetched for a given time is already an observable property with existing public API. You should be able to construct a case where, with binary search, *the wrong caption is reported* without the internal sorting step, and thus validate that it is being handled correctly without directly inspecting internal state via new public accessors.

Testing internal implementation details is an anti-pattern, and should only be done when the important properties cannot be tested via the actual client APIs. I have not seen any argument for why that would not be the case here. What scenario are you unable to test without adding a new public API?

Reply:
i didnt mean all the testing of the testing is ignored, just this test `captions are sorted correctly on initialization` since it requires the expose of the public interface so i can check if they are sorted correctly 

`List<Caption>? get sortedCaptions` without it i cant get the sorted list and without the sorted list i cant make that test

but i did add test cases for not sorted captions and tested retrieving them which works correctly and if the sorting is removed they wont pass

so i think those are enough, and no need to introduce the public interface since all of the cases are covered as you mentioned

and i did remove the public method.